### PR TITLE
Allow administrators to deploy branches of the template application

### LINKED ISSUE #17: Allow exhibits to point and upgrade to other branches (or repositories)

### PR DESCRIPTION
**Pull request by cbeer**
Fixes #17 

And add authorization controls for changing the branch configuration, limiting access to just administrative users. Previously, administrative functions were protected by "security-by-its-no-big-deal-if-someone-can", but changing branches seems a little more important.

screen shot 2015-11-13 at 9 59 49 am

### CODE REVIEW

File: app/jobs/exhibit_capistrano_deploy_job.rb [hunk 1]
```diff
@@ -14,7 +14,10 @@ def perform(request)
     return requeue_for_later(request) unless prerequisites_met?(request)
 
     Dir.chdir File.join(Rails.root, 'deploy') do
-      `EXHIBIT=#{request.slug} BRANCH=master DEPLOY_TO=#{request.exhibit_deploy_root} BUNDLE_PATH=#{Settings.capistrano_bundler_path}  BUNDLE_WITHOUT=#{Settings.capistrano_bundler_without} cap production deploy &> #{request.exhibit_deploy_root}/deploy.log`
+      request_env = "EXHIBIT=#{request.slug} BRANCH=#{request.deploy_branch} DEPLOY_TO=#{request.exhibit_deploy_root}"
+      bundle_env = "BUNDLE_PATH=#{Settings.capistrano_bundler_path} BUNDLE_WITHOUT=#{Settings.capistrano_bundler_without}"
+      log_path = "#{request.exhibit_deploy_root}/deploy.log"
+      `#{request_env} #{bundle_env} cap production deploy &> #{log_path}`
     end
```
Comment (ndushay) [inline app/jobs/exhibit_capistrano_deploy_job.rb]: :+1: I like this better - I can see what's going on a bit better.

**Comment by ndushay** [comment]
this looks good to me.  Do you feel comfortable with me merging it, or would you like @jkeck to look at it too?

**Comment by cbeer** [comment]
If you like it, ship it. Hopefully it'll be useful, particularly in the -stage environment, to let us preview, discuss, and iterate on pull requests to the template application (like the parallel indexing, or hit highlighting, or whatever)

File: app/jobs/exhibit_capistrano_deploy_job.rb (hunk 1 above)
Comment (cbeer) [inline app/jobs/exhibit_capistrano_deploy_job.rb]: Agreed. We went over the (ridiculously high) line length limit, and rubocop helpfully flagged it.

**Pull request merged**